Add section for service map (#63)
grafana/x-ray-datasource: README.md

In Insights you can see the summary table for Insights. Clicking the InsightId will take you to AWS console.

### Service map

[TODO: image of the map in grafana]

> Service map shows the same data as a service map inside X-ray console. It needs Grafana v7.4+ to use it's Node graph visualization otherwise it will show the data in a simple table.

Reviewer: I am confused. What is "Service map" in this context? Is it something in the X-ray data source in Grafana?

Author: "Service map" here means data that you get when run a "service map" query in the X-ray data source.

Reviewer: Is "Service map" used in the UI, either for the data source or the plugin? If so, please match the UI capitalization.

If not, then will it be clear to users what you mean by "service map"?

Author: Yeah wonder about that, we have Service Map in the query type:
![Screenshot from 2021-01-25 14-25-06](https://user-images.githubusercontent.com/1014802/105711736-3f8da280-5f19-11eb-8e71-54d5f8c12ee8.png)

So the first service means the query type inside the plugin, the second one means service map inside the X-ray console. Does it make sense in that case to change to `Service Map shows the same data as a service map inside X-ray console` do differentiate that?

Reviewer: Yes, that would be perfect.

> Service map shows the same data as a service map inside X-ray console. It needs Grafana v7.4+ to use it's Node graph visualization otherwise it will show the data in a simple table.

Reviewer: ```suggestion
Service map shows the same data as a service map inside X-ray console.

To display the service map:

- Use the [Node graph panel](LINK) visualization in Grafana 7.4 plus.
- Show the data in a simple table. This is the only option if the Node graph panel is unavailable.
```

Author: Not sure if this should mention the Explore? In explore in v7.4 you will automatically get the Node graph you don't need to select it.

Reviewer: Do you mean that if you click a node that it opens the trace in Explore or that the service map visualization is available in Explore?

The former, definitely. I don't know if the latter is true.

Author: So both are true. But I guess here we are talking more about how and where you can show the service map and what are the prerequisites so I wonder if there should be  

- Use the [Node graph panel](LINK) visualization in Grafana 7.4 plus.
- Use Explore in Grafana 7.4 plus.
- Show the data in a simple table. This is the only option if the Node graph panel is unavailable.
- 
Or something like that

Reviewer: Yes, but add a link to the Explore topic in the bullet.

> You can pan and zoom the view with buttons or you mouse. For details about the visualization see the Grafana docs [TODO: link].

Reviewer: I recommend deleting this paragraph and linking in line 152, above.

[TODO: image of a service with open context menu]

> Similar to X-ray root nodes, usually representing the client application are on the left. Each service in the map is represented as a circle and inside shows average time per transaction and transactions per minute. Color circle around the service also match colors and meaning from X-ray console representing percentage of requests, green being success, red for faults, yellow for errors and purple for throttled responses.

Reviewer: ```suggestion
Similar to X-ray root nodes, nodes in the service map representing the client application are on the left side of the map.

Each service in the map is represented as a circle. Numbers on the inside shows average time per transaction and transactions per minute.

A colored circle around the service also matches colors and meaning from X-ray console representing percentage of requests.

- green = success
- red = fault
- yellow = errors
- purple = throttled responses
```

> Clicking on the service or the edge will show a context menu with links additional links for quick navigation to other relevant information from X-ray. You can for example quickly navigate to list of all error traces for a particular service and then continue to specific trace.

Reviewer: ```suggestion
Click on the service or the edge to see a context menu with links additional links for quick navigation to other relevant information from X-ray. You can use the links to quickly navigate to a list of all error traces for a particular service and then continue to specific trace.
```

> For more details about the Service map see the X-ray documentation [TODO: link].

Reviewer: Isn't this the X-ray documentation? Do you mean the Node graph panel documentation?

Author: No, I meant the aws x-ray documentation. They don't seem to have specific section for the Service map but they talk a bit about how to use it in context of other features here https://docs.aws.amazon.com/xray/latest/devguide/xray-console-insights.html for example.

Reviewer: In that case, let's clarify the text.
```suggestion
For more information about the Service map, refer to the official [AWS X-ray documentation](https://docs.aws.amazon.com/xray/latest/devguide/xray-console-insights.html).
```

In Insights you can see the summary table for Insights. Clicking the InsightId will take you to AWS console.

### Service map

[TODO: image of the map in grafana]

> Service map in Grafana enables customers to view the map of their applications built using microservices architecture. Each node on the map represents a service such as an AWS Lambda function or API running on API Gateway or a DynamoDB table. With this map, customers can easily detect performance issues, or increase in error, fault or throttle rates in any of their services and dive deep into corresponding traces and root cause.

Reviewer: I recommend putting this paragraph under the heading, above the image.

This functionality needs Grafana v7.4+ to use it's Node graph visualization otherwise it will show the data in a simple table.

> You can pan and zoom the view with buttons or you mouse. For details about the visualization see the Grafana docs [TODO: link].

Reviewer: ```suggestion
You can pan and zoom the view with buttons or you mouse. For details about the visualization, refer to [Node graph panel](LINK).
```
